tracer/src/Datadog.Trace.SourceGenerators/TagsListGenerator/Sources.cs

```diff
@@ -80,27 +82,61 @@ partial class ")
             {
                 foreach (var property in tagList.MetricProperties)
                 {
+                    var tagBytes = Encoding.UTF8.GetBytes(property.TagValue);
+                    var tagArray = new string[tagBytes.Length];
+                    for (var i = 0; i < tagBytes.Length; i++)
+                    {
+                        tagArray[i] = tagBytes[i].ToString();
+                    }
+
+                    var tagByteArray = "{ " + string.Join(", ", tagArray) + " }";
+
                     sb.Append(
                            @"
-        private static readonly byte[] ")
+        // ")
                       .Append(property.PropertyName)
                       .Append(@"Bytes = Datadog.Trace.Vendors.MessagePack.StringEncoding.UTF8.GetBytes(""")
                       .Append(property.TagValue)
                       .Append(@""");");
+
+                    sb.Append(
+                           @"
+        private static readonly byte[] ")
+                      .Append(property.PropertyName)
+                      .Append(@"Bytes = new byte[] ")
+                      .Append(tagByteArray)
+                      .Append(@";");
                 }
             }
 
             if (tagList.TagProperties is not null)
             {
                 foreach (var property in tagList.TagProperties)
                 {
+                    var tagBytes = Encoding.UTF8.GetBytes(property.TagValue);
+                    var tagArray = new string[tagBytes.Length];
+                    for (var i = 0; i < tagBytes.Length; i++)
+                    {
+                        tagArray[i] = tagBytes[i].ToString();
+                    }
+
+                    var tagByteArray = "{ " + string.Join(", ", tagArray) + " }";
```

Comment:
```suggestion
                    var tagByteArray = string.Join(", ", tagArray);
```

Reply:
Oh I missed that, you're right, done in https://github.com/DataDog/dd-trace-dotnet/pull/2753/commits/9d2c4e8620171847cbe0aa374120ab261ad7eb8d